projects/ad400x/common/ad400x_bd.tcl

```diff
@@ -0,0 +1,119 @@
+
+create_bd_intf_port -mode Master -vlnv analog.com:interface:spi_master_rtl:1.0 ad400x_spi
+
+# create a SPI Engine architecture
+
+create_bd_cell -type hier spi_ad400x
+current_bd_instance /spi_ad400x
+
+  create_bd_pin -dir I -type clk clk
+  create_bd_pin -dir I -type rst resetn
+  create_bd_intf_pin -mode Master -vlnv analog.com:interface:spi_master_rtl:1.0 m_spi
+  create_bd_intf_pin -mode Master -vlnv xilinx.com:interface:axis_rtl:1.0 M_AXIS_SAMPLE
+
+  ad_ip_instance spi_engine_execution execution
+  ad_ip_parameter execution CONFIG.DATA_WIDTH $adc_resolution
+  ad_ip_parameter execution CONFIG.NUM_OF_CS 1
+  ad_ip_parameter execution CONFIG.NUM_OF_SDI 1
+
+  ad_ip_instance axi_spi_engine axi
+  ad_ip_parameter axi CONFIG.DATA_WIDTH $adc_resolution
+  ad_ip_parameter axi CONFIG.NUM_OFFLOAD 1
+
+  ad_ip_instance spi_engine_offload offload
+  ad_ip_parameter offload CONFIG.DATA_WIDTH $adc_resolution
+
+  ad_ip_instance spi_engine_interconnect interconnect
+  ad_ip_parameter interconnect CONFIG.DATA_WIDTH $adc_resolution
+
+  ad_ip_instance util_pulse_gen trigger_gen
+
+  ## to setup the sample rate of the system change the PULSE_PERIOD value
+  ## the acutal sample rate will be PULSE_PERIOD * (1/sys_cpu_clk)
+  ## fsys_cpu_clk must be 100 MHz
+  ## Default tCYC = 560 ns
+  ad_ip_parameter trigger_gen CONFIG.PULSE_PERIOD 56
+  ad_ip_parameter trigger_gen CONFIG.PULSE_WIDTH 1
+
+  if {$adc_resolution != 16} {
+    ad_ip_instance util_axis_upscale axis_upscaler
+    ad_ip_parameter axis_upscaler CONFIG.NUM_OF_CHANNELS 1
+    ad_ip_parameter axis_upscaler CONFIG.DATA_WIDTH 20
```

Comment:
$adc_resolution?

Reply:
The AD40xx family contains devices with 16, 18 and 20 bit resolution. In my understanding the EVAL-AD400xfmc board, (this name was used in the subject too, not really a typo) can have any of these devices.

Reply:
I mean use $adc_resolution instead of 20, since it can also be 18.

Reply:
Got it. Thanks!